### PR TITLE
[56048] Mockdata sync api

### PR DESCRIPTION
**Pull request by rileyanderson**
## Summary
Creates an api only available in staging that takes an ICN parameter, queriers MPI test, creates yml from response, and returns the yml in a JSON response

- Api key added to parameter store
- department-of-veterans-affairs/devops#12781
- department-of-veterans-affairs/vsp-infra-application-manifests#1729

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#56048

## Testing done

- Specs
- Tested methods inside staging rails console

## How to test:
The MPI test instance will only work on dev and staging but we can test the Mockdata::MPI::Find methods in the staging console.
- Get on a staging pod
- Start rails console: `bundle exec rails c`
- Get an ICN of a user from the test-users csv
- copy 🍝 contents of `mocked_authentication/mockdata/mpi/find.rb` file into the console
- Call `MockedAuthentication::Mockdata::MPI::Find.new(icn: icn).perform`

Route:
- You can test the route locally but it will return an error because it cant connect to MPI:
- Add api_key to `settings.local.yml`
  ```yml
  sign_in:
    mockdata_sync_api_key: {your_key}
  ```
- Make a GET request to `mocked_authentication/mpi/mockdata/:icn' with header => `Authorization: Bearer {your_key}`
- You will get a bad request error because we cant connect to MPI locally
- You can change the value of `yml` in `mockdata#show` to return something random to see a successful call 

## What areas of the site does it impact?
mock authentication

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/mocked_authentication/app/controllers/mocked_authentication/mpi/mockdata_controller.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'mockdata/mpi/find'
+
+module MockedAuthentication
+  module MPI
+    class MockdataController < MockedAuthentication::ApplicationController
+      include ActionController::HttpAuthentication::Token::ControllerMethods
+      skip_before_action :verify_authenticity_token
```
[the remark this answers is not in the document]
Comment (rileyanderson) [reply modules/mocked_authentication/app/controllers/mocked_authentication/mpi/mockdata_controller.rb:9]: CSRF tokens are not necessary unless cookies are being used for authentication. This route is protected by a Bearer token and not used by web requests

**Comment by bramleyjl** [comment]
Looks good, though apparently I don't have the AWS perms to test it on staging. Can you update the Mocked Authentication documentation with these instructions when this lands? Right now [the docs are public](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/identity/Mocked%20Authentication), so you could make sure everything here is safe to put there or maybe create a sensitive document and link to it from there.

**Review by bosawt — APPROVED** [review]
I tested by grabbing the MPI request itself and comparing it to an expected request.

Isn't really anything else I can to do test locally IMO

**Comment by bosawt** [comment]
@joeniquette Just adding you to request the authorization technique. This is just using an expected app token that we'll create, but at least wanted you to be aware. (I'm not sure if we need anything 'more secure' since this is staging and below only, the security ramifications are that if compromised, someone could possibly get an MPI response for a staging user)

File: modules/mocked_authentication/app/controllers/mocked_authentication/mpi/mockdata_controller.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (joeniquette) [reply modules/mocked_authentication/app/controllers/mocked_authentication/mpi/mockdata_controller.rb:9]: I would also add that this functionality is only for staging and the only vulnerability is on a per user basis for the forgery, if possible, and therefore would only have the ability to compromise a fake data user.

**Review by joeniquette — APPROVED** [review]
solution seems sound, going to meet later today to help with documenting.

**Review by rjohnson2011 — APPROVED** [review]

**Pull request merged**